### ISSUE TITLE
"AttributeError: NoneType object has no attribute 'wdir'" on dvc get-url call

### ISSUE THREAD

**Issue by nickdelgrosso** [iterative/dvc#3897]
## Bug Report

When calling ``dvc get-url gdrive://{mylink}`` from the command line, I am getting the following AttributeError: "NoneType' object has no attribute 'wdir'.  The full traceback is below.

I'm calling ``dvc get-url`` from outside the .dvc working directory.  A quick look at the source code suggests that this is related to output.LocalOutput being initialized with stage=None from the dependency.loads_from() function in repo.get_url.py.

### Please provide information about your setup
**Output of `dvc version`:**
```sh
$ dvc version
DVC version: 0.94.0
Python version: 3.7.6
Platform: Windows-10-10.0.18362-SP0
Binary: False
Package: pip
Supported remotes: gdrive, http, https
Filesystem type (workspace): ('NTFS', 'C:\\')
```

**Additional Information (if any):**

```
Traceback (most recent call last):
  File "d:\miniconda3\lib\site-packages\dvc\main.py", line 49, in main
    ret = cmd.run()
  File "d:\miniconda3\lib\site-packages\dvc\command\get_url.py", line 17, in run
    Repo.get_url(self.args.url, out=self.args.out)
  File "d:\miniconda3\lib\site-packages\dvc\repo\get_url.py", line 17, in get_url
    (dep,) = dependency.loads_from(None, [url])
  File "d:\miniconda3\lib\site-packages\dvc\dependency\__init__.py", line 83, in loads_from
    ret.append(_get(stage, s, info))
  File "d:\miniconda3\lib\site-packages\dvc\dependency\__init__.py", line 68, in _get
    return LocalDependency(stage, p, info)
  File "d:\miniconda3\lib\site-packages\dvc\output\local.py", line 25, in __init__
    super().__init__(stage, path, *args, **kwargs)
  File "d:\miniconda3\lib\site-packages\dvc\output\base.py", line 98, in __init__
    self.path_info = self._parse_path(remote, path)
  File "d:\miniconda3\lib\site-packages\dvc\output\local.py", line 40, in _parse_path
    p = self.stage.wdir / p
AttributeError: 'NoneType' object has no attribute 'wdir'
```

**Comment by skshetry** [comment]
Hi, @nickdelgrosso. Gdrive currently does not support `get-url`.

**Comment by efiop** [comment]
We should definitely handle it better though. Same ugly error is showing up when you `dvc get-url local-non-existent-file`. We def need to look into it.

Btw, support for gdrive will be coming along in the nearest future. We are waiting for https://github.com/iterative/PyDrive2/pull/39. So stay tuned :slightly_smiling_face:

**Comment by efiop** [comment]
@nickdelgrosso To solve it for now, we could simply check `if stage` in https://github.com/iterative/dvc/blob/master/dvc/output/local.py#L38  and assume that `wdir` is `.` otherwise. That would be in-line with https://github.com/iterative/dvc/blob/master/dvc/output/local.py#L20 . Could also pass a dummy stage in get-url instead, but that's probably the easiest way to solve it that is consisent with existing assumptions. Maybe you'd like to contribute a fix for this one too :slightly_smiling_face:

**Comment by casperdcl** [comment]
also to cross-link support for gdrive streaming: #3408

**Comment by efiop** [comment]
Ok, we need to revisit our `download` implementation. Currently it is aimed for `pull` operation, which makes it ignore errors and just log them and return int code. Our `tree.download` should raise properly instead. And should properly translate cloud-specific errors into OSErrors. E.g. missing file should make `download` raise FileNotFoundError. CC @pmrowla

**Comment by efiop** [comment]
This wasn't fixed by #3916. Reopening
